biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceBuilder.java

```diff
@@ -715,7 +715,7 @@ public void addContentCapability(URI uri, String sha256, long length, String mim
 
 	public boolean addFile(File file, URI uri) throws Exception {
 		if (uri == null)
-			uri = file.toURI();
+			uri = file.getCanonicalFile().toURI();
```

Comment:
I am not sure this is a good idea. The caller supplies the File objects and can supply canonical/absolute/relative File objects if they choose.